packages/perseus-core/src/parse-perseus-json/perseus-parsers/radio-widget.ts

```diff
@@ -47,3 +53,107 @@ export const parseRadioWidget: Parser<RadioWidget> = parseWidget(
         noneOfTheAbove: optional(constant(false)),
     }),
 );
+
+const version1 = optional(object({major: constant(1), minor: number}));
+const parseRadioWidgetV1: Parser<RadioWidget> = parseWidgetWithVersion(
+    version1,
+    constant("radio"),
+    object({
+        choices: array(
+            object({
+                content: defaulted(string, () => ""),
+                clue: optional(string),
+                correct: optional(boolean),
+                isNoneOfTheAbove: optional(boolean),
+                // deprecated
+                // There is an import cycle between radio-widget.ts and
+                // widgets-map.ts. The anonymous function below ensures that we
+                // don't refer to parseWidgetsMap before it's defined.
+                widgets: optional((rawVal, ctx) =>
+                    parseWidgetsMap(rawVal, ctx),
+                ),
+            }),
+        ),
+        hasNoneOfTheAbove: optional(boolean),
+        countChoices: optional(boolean),
+        randomize: optional(boolean),
+        multipleSelect: optional(boolean),
+        deselectEnabled: optional(boolean),
+        // deprecated
+        onePerLine: optional(boolean),
+        // deprecated
+        displayCount: optional(any),
+        // v0 props
+        // `noneOfTheAbove` is still in use (but only set to `false`).
+        noneOfTheAbove: optional(constant(false)),
+    }),
+);
+
+function migrateV1ToV2(
+    widget: ParsedValue<typeof parseRadioWidgetV1>,
+): RadioWidget {
+    const {options} = widget;
+    return {
+        ...widget,
+        version: {major: 2, minor: 0},
+        options: {
+            ...options,
+            numCorrect: deriveNumCorrect(options),
+        },
+    };
+}
+
+const version0 = optional(object({major: constant(0), minor: number}));
+const parseRadioWidgetV0: Parser<RadioWidget> = parseWidgetWithVersion(
```

Comment:
So Radio version 0 and 1 already existed, but for some reason there wasn't an existing migration for Radio. The parser got upset with me when I tried to add a migration from 1 to 2 without having one for 0 to 1.

Reply:
I didn't write a migration for v0 -> v1 because it's fairly trivial and I don't think `hasNoneOfTheAbove` is even used anymore.

```
    "1": (v0props: any): PerseusRadioWidgetOptions => {
        const {noneOfTheAbove, ...rest} = v0props;

        if (noneOfTheAbove) {
            throw new Error(
                "radio widget v0 no longer supports auto noneOfTheAbove",
            );
        }

        return {
            ...rest,
            hasNoneOfTheAbove: false,
        };
    },
```